### PR TITLE
Added LMFormatEnforcer integration

### PR DESCRIPTION
**Pull request by noamgat**
Following the discussion on the main haystack repo : https://github.com/deepset-ai/haystack/issues/6222

This PR adds the LMFormatEnforcer integration.

Note: The pypi link is to the main repository, not a dedicated haystack integration one. `LMFormatEnforcer` is a relatively small library, and all of the integrations are lazily loaded and are not part of the project's dependencies. Therefore there is no "extra baggage" by just pointing to the library itself.

### CODE REVIEW

**Review by TuanaCelik — COMMENTED** [review]
Linking to the main repo that's not only a haystack integration is totally fine. If you like, you can make the repo link go to the file where the custom component is implemented.

A couple of notes:
You could add `version: Haystack 2.0` into your frontmatter which will generate the 'Haytsack 2.0' badge on the website for you.

I will suggest some edits on certainl lines here too. Other than that, all looks good! I will maybe change the size of the image but we can see how this one looks once we merge it first.

File: integrations/lmformatenforcer.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Use the [LM Format Enforcer](https://github.com/noamgat/lm-format-enforcer)  to enforce JSON Schema / Regex output of your local models in your haystack pipelines.
+
+![Kiku](../logos/lmformatenforcer.png)
+
+## Installation
+Install the format enforcer via pip:  `pip install lm-format-enforcer`
+
+## Usage
+This integration supports both the v1 and v2 Haystack APIs:
```
Comment (TuanaCelik) [inline integrations/lmformatenforcer.md:27]: "supports both Haystack 1.x and Haystack 2.0"

File: integrations/lmformatenforcer.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+- `LMFormatEnforcerPromptNode`: A V1 `PromptNode` that activates the format enforcer.
```
Comment (TuanaCelik) [inline integrations/lmformatenforcer.md:28]: A Haystack 1.x `PromptNode that activates the format enforcer.

File: integrations/lmformatenforcer.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- `LMFormatEnforcerLocalGenerator`: A V2 generator component that activates the format enforcer.
```
Comment (TuanaCelik) [inline integrations/lmformatenforcer.md:29]: A Haystack 2.0 Generator component that activates the format enforcer

File: integrations/lmformatenforcer.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+class AnswerFormat(BaseModel):
+    first_name: str
+    last_name: str
+    year_of_birth: int
+    num_seasons_in_nba: int
+
+parser = JsonSchemaParser(AnswerFormat.schema())
+```
+### Haystack V1 Integration
```
Comment (TuanaCelik) [inline integrations/lmformatenforcer.md:51]: Haystack 1.x Integration

File: integrations/lmformatenforcer.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+{
+"first_name": "Michael",
+"last_name": "Jordan",
+"year_of_birth": 1963,
+"num_seasons_in_nba": 15
+}
+```
+For a full example, see the [example notebook](https://github.com/noamgat/lm-format-enforcer/blob/main/samples/colab_haystackv1_integration.ipynb)
+
+### Haystack V2 Integration
```
Comment (TuanaCelik) [inline integrations/lmformatenforcer.md:88]: Haystack 2.0 Integration

**Comment by noamgat** [comment]
Edited based on all of your comments.
About the logo - maybe it's time I actually make one :)

**Comment by TuanaCelik** [comment]
@noamgat LGTM, I'm sorry I didn't spot this the first round, but if possible at the very top I think it would be useful to have a short description on why this component is useful and why to use it. Something like the intro paragraph on your own repo. I think it would be useful for people who don't know about why format enforcing matrers. Once that's done, I will merge this and it will go live 🚀

Thanks for the contribution!

**Comment by noamgat** [comment]
No problem, done!

**Review by TuanaCelik — APPROVED** [review]

**Pull request merged**